lib/json5/ast.ml

```diff
@@ -0,0 +1,58 @@
+open Yojson
+type t = Yojson.Basic.t
+
+type internal =
+  | Assoc of (string * internal) list
+  | List of internal list
+  | Stringlit of string
+  | IntLit of string
+  | FloatLit of string
+  | Bool of bool
+  | Null
+  | Tuple of internal list
+  | Variant of (string * internal option)
+
+let strip_quotes s =
+  String.(sub s 1 (length s - 2))
+
+let safe_strip_quotes s =
+  if String.(get s 0 = '"')
+  then strip_quotes s
+  else s
+
+let rec to_basic : internal -> Yojson.Basic.t = function
+  | Assoc l -> `Assoc (List.map (fun (name, obj) -> (safe_strip_quotes name, to_basic obj)) l)
+  | List l -> `List (List.map to_basic l)
+  | Stringlit s -> 
+    `String (strip_quotes s)
+  | FloatLit s -> `Float (float_of_string s)
+  | IntLit s -> `Int (int_of_string s)
+  | Bool b -> `Bool b
+  | Null -> `Null
+  | Tuple _ -> raise @@ Json_error "Yojson.Basic.t do not support tuple extension"
```

Comment:
Oh, okay I didn't know about the deprecation of tuples, variants are deprecated too?

Reply:
Yes: https://github.com/ocaml-community/yojson/issues/104. Likewise for the variants: https://github.com/ocaml-community/yojson/issues/105

They are supported in JSON non-5 at the moment since we haven't yet done the work of removing them (and potentially updating `atd` in the process) since we're working out the last kinks in the 2.x release, but the plan is there.